lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -324,14 +327,39 @@ export class StandardController implements IController {
             this.initialized
         );
 
+        // Throw an error if user has set OIDCOptions without being in OIDC protocol mode
+        if(this.config.auth.protocolMode !== ProtocolMode.OIDC && 
+            this.config.auth.OIDCOptions) {
+                throw ClientConfigurationError.createCannotSetOIDCOptionsError();
+        }
+
+        let found_hash = hash;
+        
+        if(this.config.auth.protocolMode === ProtocolMode.OIDC && 
+            this.config.auth.OIDCOptions?.serverResponseType === ServerResponseType.QUERY) {
+                /**
+                 * Check from ?code to make sure we don't get a random query string
+                 * until # since some IDPs add stuff that doesn't concern MSAL after the # 
+                 */
+                const url = window.location.href;
+                if(url.indexOf("?code") > -1) {
+                    if(url.indexOf("#") > -1) {
+                        found_hash = url.substring(url.indexOf("?code") + 1, url.indexOf("#"));
```

Comment:
We have utility functions in `UrlString` (msal-common) that can handle breaking the down url into its separate parts. Please take advantage of those APIs.

Reply:
I don't think we have one that does exactly this; am I wrong about that, or should I make a utility function that will do this?

Reply:
@shylasummers try to leverage whatever you find there, if you can use functions in there to achieve parts of this then you can re-use that code, even if you do have to add a new function to wrap the steps and add the extra logic.